include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1574 @@
+
+            if (!check_path(layout, pre_t, new_pos, check_if_empty::DEST).empty())
+            {
+                layout.resize({layout.x() + 1, layout.y() + 1, 1});
+                const tile<ObstrLyt> drain{layout.x(), layout.y(), 0};
+
+                if (!check_path(layout, new_pos, drain, check_if_empty::SRC).empty())
+                {
+                    possible_positions.push_back(new_pos);
+                    count_expansions++;
+                }
+
+                layout.resize({layout.x() - 1, layout.y() - 1, 1});
+            }
+        };
+
+        // iterate diagonally
+        for (uint64_t k = 0ul; k < layout.x() + layout.y() + 1; ++k)
+        {
+            for (uint64_t x = 0ul; x < k + 1; ++x)
+            {
+                const auto y = k - x;
+                if ((pre_t.y + y) <= layout.y() && (pre_t.x + x) <= layout.x())
+                {
+                    check_tile(x, y);
+                }
+                if (count_expansions >= num_expansions)
+                {
+                    return possible_positions;
+                }
+            }
+        }
+
+        return possible_positions;
+    }
+    /**
+     * Retrieves the possible positions for a double fan-in node in the given layout, based on the positions
+     * of preceding nodes and a specified number of expansions.
+     *
+     * @param layout The layout in which to find the possible positions for a double fan-in node.
+     * @param node2pos A dictionary mapping nodes from the network to signals in the layout.
+     * @param num_expansions The maximum number of positions to be returned.
+     * @param fc A vector of nodes that precede the double fanin node.
+     * @return A vector of tiles representing the possible positions for a double fan-in node.
+     */
+    [[nodiscard]] coord_vec_type<ObstrLyt>
+    get_possible_positions_double_fanin(ObstrLyt& layout, node_dict_type<ObstrLyt, tec_nt>& node2pos,
+                                        const uint64_t num_expansions, const fanin_container<tec_nt>& fc) noexcept
+    {
+        coord_vec_type<ObstrLyt> possible_positions{};
+        possible_positions.reserve(num_expansions);
+        uint64_t count_expansions = 0ul;
+
+        const auto& pre1 = fc.fanin_nodes[0];
+        const auto& pre2 = fc.fanin_nodes[1];
+
+        const auto pre1_t = static_cast<tile<ObstrLyt>>(node2pos[pre1]);
+        const auto pre2_t = static_cast<tile<ObstrLyt>>(node2pos[pre2]);
+
+        const auto min_x = std::max(pre1_t.x, pre2_t.x) + (pre1_t.x == pre2_t.x ? 1 : 0);
+        const auto min_y = std::max(pre1_t.y, pre2_t.y) + (pre1_t.y == pre2_t.y ? 1 : 0);
+
+        // check if path from previous tiles to new tile and from new tile to drain exist
+        auto check_tile = [&](uint64_t x, uint64_t y)
+        {
+            const tile<ObstrLyt> new_pos{min_x + x, min_y + y, 0};
+
+            const auto path_1 = check_path(layout, pre1_t, new_pos, check_if_empty::DEST);
```

Comment:
If there is no `path_2`, we might as well rename this variable to `path` or `p`.